### PR TITLE
Remove deprecated ESIPP beta annotations

### LINKED ISSUE #50187: TODO: remove deprecated ESIPP beta annotations in 1.8
[TODO: The beta annotations have been deprecated, remove them when we release k8s 1.8.](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/annotation_key_constants.go#L98)

As we are on the 1.8 train now, is it time to remove these deprecated annotations? I have already made the cleanup on my local branch. I can send a PR once I get the ticket :)

/kind cleanup
/sig network
/assign
/cc @MrHohn

### PR DESCRIPTION
**Pull request by xiangpengzhao**
**What this PR does / why we need it**:
Remove deprecated ESIPP beta annotations.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #50187

**Special notes for your reviewer**:
/assign @MrHohn
/sig network

**Release note**:

```release-note
Beta annotations `service.beta.kubernetes.io/external-traffic` and `service.beta.kubernetes.io/healthcheck-nodeport` have been removed. Please use fields `service.spec.externalTrafficPolicy` and `service.spec.healthCheckNodePort` instead.
```

### CODE REVIEW

**Comment by xiangpengzhao** [comment]
Maybe we want a release note for this PR as it removes some deprecated things?

**Comment by MrHohn** [comment]
Thanks for the PR, will review it soon.

> Maybe we want a release note for this PR as it removes some deprecated things?

Yep, maybe something like:
```
Beta annotations `service.beta.kubernetes.io/external-traffic` and `service.beta.kubernetes.io/healthcheck-nodeport` have been deprecated. Please use `service.spec.externalTrafficPolicy` and `service.spec.healthCheckNodePort` instead.
```

/ok-to-test

**Comment by xiangpengzhao** [comment]
@MrHohn thanks for your suggestion on release note. I modify the release-note block as below.

```
Beta annotations `service.beta.kubernetes.io/external-traffic` and `service.beta.kubernetes.io/healthcheck-nodeport` have been removed. Please use fields `service.spec.externalTrafficPolicy` and `service.spec.healthCheckNodePort` instead.
```

Note that I use `removed` instead of `deprecated`. Because I think `deprecated` things are still supported for compatibility, but `removed` things aren't. For this PR, these annotations won't be supported anymore.

Dose this make sense?

File: pkg/api/service/util.go [hunk 1]
```diff
@@ -104,45 +87,16 @@ func NeedsHealthCheck(service *api.Service) bool {
 
 // GetServiceHealthCheckNodePort Return health check node port for service, if one exists
 func GetServiceHealthCheckNodePort(service *api.Service) int32 {
```
Comment (MrHohn) [inline pkg/api/service/util.go]: Do you think it worth to get rid of this function? This is essentially checking `service.Spec.HealthCheckNodePort`. As the beta API is removed we don't need this abstraction layer anymore.

File: pkg/api/service/util.go [hunk 2]
```diff
@@ -104,45 +87,16 @@ func NeedsHealthCheck(service *api.Service) bool {
...
-			glog.Errorf("Failed to parse annotation %v: %v", api.BetaAnnotationHealthCheckNodePort, err)
-			return 0
-		}
-		return int32(p)
-	}
 	return service.Spec.HealthCheckNodePort
 }
 
 // ClearExternalTrafficPolicy resets the ExternalTrafficPolicy field.
 func ClearExternalTrafficPolicy(service *api.Service) {
```
Comment (MrHohn) [inline pkg/api/service/util.go]: The same comment here.

File: pkg/api/service/util.go [hunk 3]
```diff
@@ -104,45 +87,16 @@ func NeedsHealthCheck(service *api.Service) bool {
...
-	if _, ok := service.Annotations[api.BetaAnnotationExternalTraffic]; ok {
-		delete(service.Annotations, api.BetaAnnotationExternalTraffic)
-		return
-	}
 	service.Spec.ExternalTrafficPolicy = api.ServiceExternalTrafficPolicyType("")
 }
 
 // SetServiceHealthCheckNodePort sets the given health check node port on service.
 // It does not check whether this service needs healthCheckNodePort.
 func SetServiceHealthCheckNodePort(service *api.Service, hcNodePort int32) {
```
Comment (MrHohn) [inline pkg/api/service/util.go]: The same.

File: pkg/api/service/util.go (hunk 1 above)
Comment (MrHohn) [inline pkg/api/service/util.go]: And so we could remove those unit tests as well :) WDYT?

**Comment by MrHohn** [comment]
> Note that I use removed instead of deprecated. Because I think deprecated things are still supported for compatibility, but removed things aren't. For this PR, these annotations won't be supported anymore.
>
> Dose this make sense?

Thanks, the release note SGTM.

**Review by MrHohn — COMMENTED** [review]
Codes look good, couple minor comments :)

File: pkg/api/service/util.go (hunk 1 above)
Comment (xiangpengzhao) [inline pkg/api/service/util.go]: Agreed. I was also thinking to remove these helpers when I first hit them. As you said, they're not needed anymore. Will address this. Thanks!

**Comment by xiangpengzhao** [comment]
@MrHohn inline comments addressed. I sent a separate commit for easy review. Feel free to tell me if I need to squash them after your review. Thanks!

**Comment by MrHohn** [comment]
Thanks, this looks great :)
/lgtm

/assign @thockin 
for approval.

**Pull request merged**